### PR TITLE
[ADF-604] Upgrade @angular/material to latest version

### PR DESCRIPTION
**Pull request by DenisVuyka**
**Please check if the PR fulfills these requirements**
```
[ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
[ ] Tests for the changes have been added (for bug fixes / features)
[ ] Docs have been added / updated (for bug fixes / features)
```

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] Documentation
[ ] Other... Please describe:
```

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

- upgrade to @angular/material beta 6
- migrate Activiti Form to @angular/material Tab component
- deprecate old MDL code and docs 

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: ng2-components/ng2-alfresco-core/src/services/notification.service.spec.ts [hunk 1]
```diff
@@ -43,33 +48,29 @@ describe('NotificationService', () => {
...
+        expect(document.querySelector('snack-bar-container')).not.toBeNull();
     });
 
-    describe('openSnackMessageAction', () => {
 
-        it('should open a message notification bar with action', (done) => {
-            let promise = fixture.componentInstance.sendMessageAction();
-            promise.afterDismissed().subscribe(() => {
-                done();
-            });
```
Comment (eromano) [inline ng2-components/ng2-alfresco-core/src/services/notification.service.spec.ts]: Why this test is excludes?

Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-core/src/services/notification.service.spec.ts]: This test executes locally but fails on Travis due to some configuration problems I can't identify

Comment (eromano) [inline ng2-components/ng2-alfresco-core/src/services/notification.service.spec.ts]: Can you try to make it works?

Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-core/src/services/notification.service.spec.ts]: I tried several times, but every experiment requires pushing content to Travis and waiting for results. All tests work fine on local machine.

Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-core/src/services/notification.service.spec.ts]: @eromano managed to reproduce locally, will try to fix the test

**Pull request merged**